mlx_graphs/nn/message_passing.py

```diff
@@ -25,72 +27,89 @@ def __init__(self, aggr="add"):
         super().__init__()
 
         self.aggr = aggr
+        self.num_nodes = None
 
     def __call__(self, x: mx.array, edge_index: mx.array, **kwargs: Any):
         raise NotImplementedError
 
-    def propagate(self, x: mx.array, edge_index: mx.array, **kwargs: Any) -> mx.array:
+    def propagate(
+        self,
+        node_features: Union[mx.array, Tuple[mx.array, mx.array]],
+        edge_index: mx.array,
+        message_kwargs: Optional[Dict] = {},
+        aggregate_kwargs: Optional[Dict] = {},
+        update_kwargs: Optional[Dict] = {},
```

Comment:
I know this is how also PyG implements it, but is there a better way to expose them?